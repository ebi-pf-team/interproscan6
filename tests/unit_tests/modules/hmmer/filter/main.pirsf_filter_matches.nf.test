nextflow_process {

    name "Test Process PIRSF_FILTER_MATCHES"
    script "interproscan/modules/hmmer/filter/main.nf"
    process "PIRSF_FILTER_MATCHES"

    stage {
        symlink "interproscan/scripts/members/pirsf/filter_ips6_hits.py"
        symlink "tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/hmmer_parsed_pirsf_out.json"
        symlink "tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/pirsf_dtbl"
        symlink "tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/pirsf.dat"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/hmmer_parsed_pirsf_out.json")
                input[1] = file("meta/tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/pirsf_dtbl")
                input[2] = [
                    "../../../meta/tests/unit_tests/test_inputs/hmmer_filter_matches/pirsf/pirsf.dat"
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
