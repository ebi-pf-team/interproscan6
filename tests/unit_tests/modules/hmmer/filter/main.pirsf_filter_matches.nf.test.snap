{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmer_parsed_pirsf_out.json.post.processed.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T15:50:53.92941"
    }
}